src/System.Private.CoreLib/src/System/Diagnostics/Eventing/TraceLogging/TraceLoggingEventHandleTable.cs

```diff
@@ -0,0 +1,61 @@
+// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+// See the LICENSE file in the project root for more information.
+using System.Runtime.CompilerServices;
+using System.Threading;
+
+namespace System.Diagnostics.Tracing
+{
+#if FEATURE_PERFTRACING
+    /// <summary>
+    /// Per-EventSource data structure for caching EventPipe EventHandles associated with TraceLogging events.
+    /// </summary>
+    internal sealed class TraceLoggingEventHandleTable
+    {
+        private const int DefaultLength = 10;
+        private IntPtr[] m_innerTable;
+        private object m_innerTableUpdateLock = new object();
+
+        internal TraceLoggingEventHandleTable()
+        {
+            m_innerTable = new IntPtr[DefaultLength];
+        }
+
+        internal IntPtr this[int eventID]
+        {
+            get
+            {
+                IntPtr ret = IntPtr.Zero;
+                IntPtr[] innerTable = Volatile.Read(ref m_innerTable);
+
+                if (eventID >= 0 && eventID < innerTable.Length)
```

Comment:
In what situation would eventID be < 0?

Reply:
This can occur early in the TraceLogging pipeline before an EventID is assigned.  It generally shouldn't occur here, but the goal is to avoid throwing if the bounds check fails.  We'll return IntPtr.Zero here and then when the event is sent to EventPipe it will be discarded because the event handle is invalid.

Reply:
Ok.  If you wanted, you could make this:
```C#
if ((uint)eventID < (uint)innerTable.Length)
```